### PR TITLE
[test](array function)add array_range function test

### PR DESCRIPTION
**Pull request by zy-kkk**
# Proposed changes

Issue Number: close #xxx

## Problem summary

Describe your changes.

## Checklist(Required)

1. Does it affect the original behavior: 
    - [ ] Yes
    - [ ] No
    - [ ] I don't know
2. Has unit tests been added:
    - [x] Yes
    - [ ] No
    - [ ] No Need
3. Has document been added or modified:
    - [ ] Yes
    - [ ] No
    - [ ] No Need
4. Does it need to update dependencies:
    - [ ] Yes
    - [ ] No
5. Are there any changes that cannot be rolled back:
    - [ ] Yes (If Yes, please explain WHY)
    - [ ] No

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by hello-stephen** [comment]
TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 34.67 seconds
 load time: 441 seconds
 storage size: 17181053104 Bytes
 https://doris-community-test-1308700295.cos.ap-hongkong.myqcloud.com/tmp/20221109124913_clickbench_pr_42804.html

File: regression-test/suites/query_p0/sql_functions/array_functions/test_array_functions.groovy [hunk 1]
```diff
@@ -70,4 +70,32 @@ suite("test_array_functions") {
...
+              `k3` int(11) NULL COMMENT ""
+            ) ENGINE=OLAP
+            DUPLICATE KEY(`k1`)
+            DISTRIBUTED BY HASH(`k1`) BUCKETS 1
+            PROPERTIES (
+            "replication_allocation" = "tag.location.default: 1",
+            "storage_format" = "V2"
+            )
+        """
+    sql """ INSERT INTO ${tableName2} VALUES(1,3,5) """
```
Comment (cambyzju) [inline regression-test/suites/query_p0/sql_functions/array_functions/test_array_functions.groovy:89]: Could you also add some abnormal cases? such as:
1. end > start;
2. step > (end - start);
3. negative start, end and step;

Comment (zy-kkk) [reply regression-test/suites/query_p0/sql_functions/array_functions/test_array_functions.groovy:90]: done

**Review by cambyzju — APPROVED** [review]
LGTM

**Pull request merged**